### PR TITLE
Fix periodic benchmarks and remove version cap on `asv`

### PR DESCRIPTION
**Pull request by agriyakhetarpal**
# Description

The scheduled benchmarks suite was failing to run because the version was not pinned to 0.5.1. This PR resolves both benchmarks workflows and allows the benchmarks to run

## Type of change

Please add a line in the relevant section of [CHANGELOG.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CHANGELOG.md) to document the change (include PR #) - note reverse order of PR #s. If necessary, also add to the list of breaking changes.

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (back-end change that speeds up the code)
- [x] Bug fix (non-breaking change which fixes an issue)

# Key checklist:

- [ ] No style issues: `$ pre-commit run` (or `$ nox -s pre-commit`) (see [CONTRIBUTING.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CONTRIBUTING.md#installing-and-using-pre-commit) for how to set this up to run automatically when committing locally, in just two lines of code)
- [ ] All tests pass: `$ python run-tests.py --all` (or `$ nox -s tests`)
- [ ] The documentation builds: `$ python run-tests.py --doctest` (or `$ nox -s doctests`)

You can run integration tests, unit tests, and doctests together at once, using `$ python run-tests.py --quick` (or `$ nox -s quick`).

## Further checks:

- [ ] Code is commented, particularly in hard-to-understand areas
- [ ] Tests added that prove fix is effective or that feature works

### CODE REVIEW

File: .github/workflows/benchmark_on_push.yml [hunk 1]
```diff
@@ -49,7 +53,8 @@ jobs:
           HEAD_COMMIT=$(git rev-parse HEAD)
           echo $BASE_COMMIT | tee commits_to_compare.txt
           echo $HEAD_COMMIT | tee -a commits_to_compare.txt
-          asv run HASHFILE:commits_to_compare.txt --m "GitHubRunner" --show-stderr --strict -v
+          asv run HASHFILE:commits_to_compare.txt --m "GitHubRunner" --show-stderr -v
```
Comment (Saransh-cpp) [inline .github/workflows/benchmark_on_push.yml:56]: Re: removal of `--strict`

Do we want this workflow to pass even if the benchmarks fail? `asv` exits with a non-zero return code only if `--strict` is present. If we remove this, the workflow will always pass.

File: .github/workflows/periodic_benchmarks.yml [hunk 2]
```diff
@@ -20,26 +20,33 @@ jobs:
...
+          python -m pip install asv
+
+      - name: Install SuiteSparse and SUNDIALS
         run: python scripts/install_KLU_Sundials.py
+
       - name: Run benchmarks
         run: |
           asv machine --machine "GitHubRunner"
-          asv run --machine "GitHubRunner" NEW --show-stderr --strict -v
+          asv run --machine "GitHubRunner" NEW --show-stderr -v
```
Comment (Saransh-cpp) [inline .github/workflows/periodic_benchmarks.yml:45]: Ditto

File: .github/workflows/benchmark_on_push.yml [hunk 3]
```diff
@@ -18,16 +18,20 @@ jobs:
...
           sudo apt-get update
           sudo apt install gfortran gcc libopenblas-dev
+
       - name: Install python dependencies
-      # Pin asv==0.5.1 to fix failing benchmarks. Related to https://github.com/airspeed-velocity/asv/issues/1323
         run: |
-          python -m pip install --upgrade pip wheel setuptools virtualenv asv==0.5.1 wget cmake casadi numpy
-      - name: Install SuiteSparse and Sundials
+          python -m pip install --upgrade pip wheel setuptools virtualenv wget cmake casadi numpy virtualenv packaging
+          python -m pip install asv[virtualenv]
```
Comment (Saransh-cpp) [inline .github/workflows/benchmark_on_push.yml:30]: We can get rid of `virtualenv` from one of these lines.

File: .github/workflows/periodic_benchmarks.yml [hunk 4]
```diff
@@ -20,26 +20,33 @@ jobs:
...
         run: |
           sudo apt-get update
           sudo apt-get install gfortran gcc libopenblas-dev
+
       - name: Install python dependencies
         run: |
-          python -m pip install --upgrade pip wheel setuptools virtualenv asv wget cmake casadi numpy
-      - name: Install SuiteSparse and Sundials
+          python -m pip install --upgrade pip wheel setuptools virtualenv wget cmake casadi numpy virtualenv packaging
+          python -m pip install asv[virtualenv]
```
Comment (Saransh-cpp) [inline .github/workflows/periodic_benchmarks.yml:37]: Ditto!

**Review by Saransh-cpp — CHANGES_REQUESTED** [review]
Thanks, @agriyakhetarpal! Some comments below -

File: .github/workflows/benchmark_on_push.yml (hunk 1 above)
Comment (agriyakhetarpal) [reply .github/workflows/benchmark_on_push.yml:56]: They deprecated the `--strict` command-line flag in `0.6` so the benchmarks will always fail if they are destined to ([changelog](https://asv.readthedocs.io/en/v0.6.1/changelog.html#id5))

File: .github/workflows/benchmark_on_push.yml (hunk 3 above)
Comment (agriyakhetarpal) [reply .github/workflows/benchmark_on_push.yml]: Done

File: .github/workflows/benchmark_on_push.yml (hunk 1 above)
Comment (Saransh-cpp) [reply .github/workflows/benchmark_on_push.yml:56]: Oh nice! Thanks for the clarification!

**Review by Saransh-cpp — APPROVED** [review]

**Pull request merged**